// @slidev/cli/dist/chunk-DRWYFCYG.mjs
import {
  ViteSlidevPlugin,
  mergeViteConfigs
} from "./chunk-ZATTXN6U.mjs";

// node/commands/server.ts
import { join } from "node:path";
import process from "node:process";
import { createServer as createViteServer, mergeConfig } from "vite";
async function createServer(options, viteConfig = {}, serverOptions = {}) {
  process.env.EDITOR = process.env.EDITOR || "code";
  const config = await mergeViteConfigs(
    options,
    viteConfig,
    {
      root: options.userRoot,
      optimizeDeps: {
        entries: [
          join(options.clientRoot, "main.ts")
        ]
      }
    },
    "serve"
  );
  const server = await createViteServer(
    mergeConfig(
      config,
      {
        plugins: [
          await ViteSlidevPlugin(options, config.slidev || {}, serverOptions)
        ],
        define: {
          // Fixes Vue production mode breaking PDF Export #1245
          __VUE_PROD_DEVTOOLS__: JSON.stringify(true)
        }
      }
    )
  );
  return server;
}

export {
  createServer
};
